xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecStarfish.cpp

```diff
@@ -0,0 +1,555 @@
+/*
+ *  Copyright (C) 2023 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "DVDVideoCodecStarfish.h"
+
+#include "CompileInfo.h"
+#include "DVDCodecs/DVDFactoryCodec.h"
+#include "ServiceBroker.h"
+#include "cores/VideoPlayer/Buffers/VideoBuffer.h"
+#include "cores/VideoPlayer/Interface/DemuxCrypto.h"
+#include "cores/VideoPlayer/Interface/TimingConstants.h"
+#include "cores/VideoPlayer/VideoRenderers/RenderManager.h"
+#include "media/decoderfilter/DecoderFilterManager.h"
+#include "messaging/ApplicationMessenger.h"
+#include "settings/Settings.h"
+#include "settings/SettingsComponent.h"
+#include "utils/BitstreamConverter.h"
+#include "utils/CPUInfo.h"
+#include "utils/JSONVariantWriter.h"
+#include "utils/TimeUtils.h"
+#include "utils/log.h"
+#include "windowing/wayland/WinSystemWaylandWebOS.h"
+
+#include <memory>
+#include <vector>
+
+using namespace KODI::MESSAGING;
+
+CDVDVideoCodecStarfish::CDVDVideoCodecStarfish(CProcessInfo& processInfo)
+  : CDVDVideoCodec(processInfo), m_starfishMediaAPI(std::make_unique<StarfishMediaAPIs>())
+{
+}
+
+CDVDVideoCodecStarfish::~CDVDVideoCodecStarfish()
+{
+  Dispose();
+}
+
+std::unique_ptr<CDVDVideoCodec> CDVDVideoCodecStarfish::Create(CProcessInfo& processInfo)
+{
+  return std::make_unique<CDVDVideoCodecStarfish>(processInfo);
+}
+
+bool CDVDVideoCodecStarfish::Register()
+{
+  CDVDFactoryCodec::RegisterHWVideoCodec("starfish_dec", &CDVDVideoCodecStarfish::Create);
+  return true;
+}
+
+std::atomic<bool> CDVDVideoCodecStarfish::ms_instanceGuard(false);
+
+bool CDVDVideoCodecStarfish::Open(CDVDStreamInfo& hints, CDVDCodecOptions& options)
+{
+  // allow only 1 instance here
+  if (ms_instanceGuard.exchange(true))
+  {
+    CLog::Log(LOGERROR, "CDVDVideoCodecStarfish::Open - InstanceGuard locked");
```

Comment:
Yes this happens when starting a video when there is already one playing. If we didn't guard this here playback would fail for the next video because there would be 2 videos trying to display on the same exported window